### PR TITLE
Delete PVCs from auto provisioned Jaeger at end of test

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

Delete PVCs at the end of tests which create Kafka instances with persistent storage.   This should not strictly be necessary, as the PVCs should be deleted when the namespace is deleted.  

However, we have had times where a subsequent test gets a panic, particularly because of port-forwarding problems, and then the namespace doesn't get deleted.  This can cause problems with some of our clusters which do not have a lot of resources for volumes.

### CODE REVIEW

File: test/e2e/utils.go [hunk 1]
```diff
@@ -414,3 +414,14 @@ func createSecret(secretName, secretNamespace string, secretData map[string][]by
 	WaitForSecret(secretName, secretNamespace)
 	return createdSecret
 }
+
+func deletePersistentVolumeClaims(namespace string) {
+	pvcs, err := fw.KubeClient.CoreV1().PersistentVolumeClaims(kafkaNamespace).List(metav1.ListOptions{})
```
Comment (jpkrohling) [inline test/e2e/utils.go]: What happens to the volumes? Are they deleted automatically?

Comment (kevinearls) [reply test/e2e/utils.go]: Normally they are deleted when the namespace is deleted, but not when the Kafka instance is deleted.  Deleting them in this function just covers some edge cases where the namespace for a test may not get deleted because a subsequent test panics, or something of that sort.  

This has caused us a few problems on internal CI.

Comment (kevinearls) [reply test/e2e/utils.go]: @jpkrohling Yes, if you delete the pvc the pv is deleted along with it.
